modules/services/holo-auth-client.nix

```
@@ -31,27 +31,48 @@ in
       wantedBy = [ "multi-user.target" ];
 
       script = ''
-        zerotier_status() {
-          zerotier-cli -j listnetworks | jq -r .[0].status
-        }
-
-        sleep 10
-
-        while [ "$(zerotier_status)" = "REQUESTING_CONFIGURATION" ]; do
-          sleep 1
+        #
+        # Phase 1: locate `hpos-state.json`; export HPOS_STATE_PATH
+        # NOTE: This *requires* us to be able to communicate via the console, hence the
+        # require/after for systemd-logind.service
+        #
+        echo >&2 "Running HPOS init..."
+        while ! HPOS_STATE_PATH=$(hpos-init) || [ -z "$HPOS_STATE_PATH" ]; do
+          # Perhaps receiving/parsing/saving hpos-state.json failed/empty; try again...
+          echo >&2 "Failed to acheive HPOS Init; retrying..."
         done
+        export HPOS_STATE_PATH
+        echo >&2 "HPOS Init successful, with path $HPOS_STATE_PATH"
 
-        export HPOS_STATE_PATH=$(hpos-init)
-
+        #
+        # Phase 2: Derive Holochain keys; export HOLO_PUBLIC_KEY
+        #
         mkdir -p /var/lib/holochain-conductor
         cd /var/lib/holochain-conductor
 
-        hpos-state-derive-keystore < $HPOS_STATE_PATH > holo-keystore 2> holo-keystore.pub
+        echo >&2 "Deriving Holo keystore to $HPOS_STATE_PATH/holo-keystore..."
+        if ! hpos-state-derive-keystore < $HPOS_STATE_PATH > holo-keystore 2> holo-keystore.pub; then
+          echo >&2 "Failed to derive keys from $HPOS_STATE_PATH"
+          exit 1
+        fi
         export HOLO_PUBLIC_KEY=$(cat holo-keystore.pub)
```

Without this loop, failure to derive the keystore are silently ignored, without logging.